lib/smart_city/data.ex

```diff
@@ -123,6 +123,23 @@ defmodule SmartCity.Data do
     e -> {:error, e}
   end
 
+  def new(%{
+      dataset_id: dataset_id,
+      operational: operational,
+      payload: payload,
+      _metadata: metadata
+  }) do
+    %{
+      dataset_id: dataset_id,
+      operational: operational,
+      payload: payload,
+      _metadata: metadata,
+      ingestion_id: nil,
+      extraction_start_time: nil
```

Comment:
If the ids are provided, could this nil them out?

Reply:
Also, will this scale to future schema changes?

Reply:
It won't, because Elixir pattern matches in order defined in the file. I've added a test which verifies this and will inform later developers if they break that functionality.

This could be scalable in a couple ways. First, by making the optional fields required once we've done the transition in the rest of the codebase. Then the extra constructor can be removed safely, and future schema changes can follow the same expand-contract pattern. Second, if we wanted to keep the fields optional, future changes could just have an extra line in this function providing a default for that new optional field.

I'm expecting to take the first approach, since we do want these fields to be required once the system supports them.

Reply:
Sounds like it should work. Adding new fields to this should be rare.